### PR TITLE
T1021 Remote services

### PR DESCRIPTION
**Pull request by VakarisZ**
# Feature
> Implemented T1021 Remote services attack technique
image

### CODE REVIEW

File: monkey/monkey_island/cc/services/attack/technique_reports/T1021.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    def get_report_data():
+        attempts = []
+        if mongo.db.telemetry.count_documents(T1021.scanned_query):
+            attempts = list(mongo.db.telemetry.aggregate(T1021.query))
+            if attempts:
+                status = ScanStatus.USED.value
+                for result in attempts:
+                    result['successful_creds'] = []
+                    for attempt in result['attempts']:
+                        result['successful_creds'].append(T1110.parse_creds(attempt))
```
Comment (itaymmguardicore) [inline monkey/monkey_island/cc/services/attack/technique_reports/T1021.py]: move `parse_creds` and its helper functions to a new file.
It doesn't make sense to import from T1110

File: monkey/monkey_island/cc/services/attack/technique_reports/T1021.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    # Gets data about brute force attempts
+    query = [{'$match': {'telem_category': 'exploit',
+                         'data.attempts': {'$not': {'$size': 0}}}},
+             {'$project': {'_id': 0,
+                           'machine': '$data.machine',
+                           'info': '$data.info',
+                           'attempt_cnt': {'$size': '$data.attempts'},
+                           'attempts': {'$filter': {'input': '$data.attempts',
+                                                    'as': 'attempt',
+                                                    'cond': {'$and': [{'$eq': ['$$attempt.result', True]},
```
Comment (itaymmguardicore) [inline monkey/monkey_island/cc/services/attack/technique_reports/T1021.py]: why do you require SSH/password? Using hashes is also legitimate in this context.

File: monkey/monkey_island/cc/services/attack/technique_reports/T1021.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                                                                      {'$or': [{'$ne': ['$$attempt.password', '']},
+                                                                               {'$ne': ['$$attempt.ssh_key', '']}]}]
+                                                             }
+                                                    }
+                                        }
+                           }
+              }]
+
+    scanned_query = {'telem_category': 'exploit',
```
Comment (itaymmguardicore) [inline monkey/monkey_island/cc/services/attack/technique_reports/T1021.py]: Same here

File: monkey/monkey_island/cc/services/attack/technique_reports/T1021.py (hunk 2 above)
Comment (VakarisZ) [reply monkey/monkey_island/cc/services/attack/technique_reports/T1021.py]: I followed mitre definition by the book. "An adversary may use Valid Accounts to log into a service specifically designed to accept remote connections, such as telnet, SSH, and VNC." Valid accounts means valid key/password, not PTH.

Comment (VakarisZ) [reply monkey/monkey_island/cc/services/attack/technique_reports/T1021.py]: Confirm that you want to change that and I'll change

Comment (itaymmguardicore) [reply monkey/monkey_island/cc/services/attack/technique_reports/T1021.py]: I don't see anything in the description which disqualifies hashes from this technique. Let's do it

File: monkey/monkey_island/cc/services/attack/technique_reports/T1021.py (hunk 3 above)
Comment (itaymmguardicore) [reply monkey/monkey_island/cc/services/attack/technique_reports/T1021.py]: Let's also do it here

**Pull request merged**